pkg/errors/grpc.go

```diff
@@ -50,6 +50,9 @@ func FromGRPCStatus(status *status.Status) Error {
 		err.details = detailIfaces
 	}
 	if details != nil {
+		if err.code == 0 {
```

Comment:
This is in the wrong place. Setting things from `ErrorDetails` into `Error` is done in `setErrorDetails`.

Reply:
This should be set only if not already present in `status`.
The behavior you propose it confusing (see https://github.com/TheThingsNetwork/lorawan-stack/pull/976#discussion_r302411445), since `setErrorDetails` then both sets if not already set and sets if present in the details
A possible solution is to just remove the `setErrorDetails` altogether and simply set fields directly
@johanstokking what solution do you propose?

Reply:
The reason it's so confusing is that we're now creating an explicit code field while we previously derived the code from the thing that contained the error (the gRPC/HTTP status code).

The `setErrorDetails` func sets error fields from the error details, and that's exactly what we're doing here, so this code belongs in there.

I'm fine with making the logic the same as the existing behavior of `setErrorDetails`, so "if the ErrorDetails has a non-zero error code, set it in Error". That way the ErrorDetails becomes leading, with fallback to the old behavior.